src/frame_graph/frame_graph.hpp

```diff
@@ -21,14 +21,44 @@
 #define FG_MAX_PERFORMANCE
 #endif
 
-#define EXPAND(x) x // Because msvc handles the preprocessor differently
-#define FG_DEPS(N, ...) EXPAND(FG_DEPS##N(__VA_ARGS__))
-#define FG_DEPS1(A) { typeid(A) }
-#define FG_DEPS2(A, B) { typeid(A), typeid(B) }
-#define FG_DEPS3(A, B, C) { typeid(A), typeid(B), typeid(C) }
-#define FG_DEPS4(A, B, C, D) { typeid(A), typeid(B), typeid(C), typeid(D) }
-#define FG_DEPS5(A, B, C, D, E) { typeid(A), typeid(B), typeid(C), typeid(D), typeid(E) }
-#define FG_DEPS6(A, B, C, D, E, F) { typeid(A), typeid(B), typeid(C), typeid(D), typeid(E), typeid(F) }
+namespace fg_deps
```

Comment:
I choose macro's to reduce the initialization time. I don't mind changing it to template magic. But this approach is far to convoluted. This can be simplified a lot. Also I'm not a fan of the style used here. Might be easiest to revert these changes and make a new pull request with a better approach.

Reply:
(I think it would be easier to revert since you need to go to all tasks again to change it. so might be good to move that to a different commit so I can merge this.)

Reply:
I'll revert it and put it into a separate PR. There's not a lot of additional initialization time using this technique, if I make it inline it should be relatively fast; and it only initializes it once and then caches it. It doesn't require the user to specify how many arguments are used (avoiding user errors) and is infinitely expandable (if for some reason there would be more than 6 dependencies; which would probably never happen). 
```cpp
	static fg_deps::type deps = fg_deps::make<args...>();
	return deps;
```
^^ it is ran once and then just uses the vector created

Reply:
https://github.com/TeamWisp/WispRenderer/issues/340 moved to an issue instead. It's removed now

Reply:
Yes I get it only runs once. But the way you wrote that code is highly convoluted as well which I why I don't like it.